include/deal.II/grid/reference_cell.h

```diff
@@ -1007,7 +1007,7 @@ ReferenceCell::n_lines() const
     return 12;
 
   Assert(false, ExcNotImplemented());
```

Comment:
```suggestion
  AssertThrow(false, ExcNotImplemented());
```
Shall we use an `AssertThrow` here to make sure that this part is never reached in both `debug` and `release` mode? What do you think?

Reply:
We don't use at such places `AssertThrow`.

Reply:
I agree; while we expect the code to not be reachable, the compiler will not know that and still generates the code also in release mode. Having a smaller function content makes the likelihood of the compiler to inline the function higher, making it a win overall. We have the debug mode for ensuring consistency of data.